src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -755,4 +774,58 @@ private  ApiResponse returnErrorMsg(String error, HttpStatus type, ApiResponse r
         response.setMessage(status);
         return response;
     }
+
+
+    @Override
+    public ApiResponse uploadFile(MultipartFile mFile) {
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.DISCUSSION_UPLOAD_FILE);
+
+        File file = null;
+        try {
+            file = new File(System.currentTimeMillis() + "_" + mFile.getOriginalFilename());
+
+            file.createNewFile();
+            // Use try-with-resources to ensure FileOutputStream is closed
+            try (FileOutputStream fos = new FileOutputStream(file)) {
+                fos.write(mFile.getBytes());
+            }
+            String yearMonth = LocalDate.now().format(DateTimeFormatter.ofPattern("yyyy-MM"));
+            String uploadFolderPath = cbServerProperties.getDiscussionCloudFolderName() + "/" + yearMonth;
+            return uploadFile(file, uploadFolderPath, cbServerProperties.getDiscussionContainerName());
+        } catch (Exception e) {
+            log.error("Failed to upload file. Exception: ", e);
+            response.getParams().setStatus(Constants.FAILED);
+            response.getParams().setErrMsg("Failed to upload file. Exception: " + e.getMessage());
+            response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+            return response;
+        } finally {
+            if (file != null && file.exists()) {
+                file.delete();
+            }
+        }
+    }
+
+    public ApiResponse uploadFile(File file, String cloudFolderName, String containerName) {
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.UPLOAD_FILE);
+        try {
+            String objectKey = cloudFolderName + "/" + file.getName();
+            String url = storageService.upload(containerName, file.getAbsolutePath(),
+                    objectKey, Option.apply(false), Option.apply(1), Option.apply(5), Option.empty());
+            Map<String, String> uploadedFile = new HashMap<>();
+            uploadedFile.put(Constants.NAME, file.getName());
+            uploadedFile.put(Constants.URL, url);
+            response.getResult().putAll(uploadedFile);
+            return response;
+        } catch (Exception e) {
+            log.error("Failed to upload file. Exception: ", e);
+            response.getParams().setStatus(Constants.FAILED);
+            response.getParams().setErrMsg("Failed to upload file. Exception: " + e.getMessage());
+            response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+            return response;
+        } finally {
+            if (file != null) {
+                file.delete();
+            }
```

Comment:
this is not required... file is not a local variable for this method

Reply:
removed